src/radix_connect/mobile/session/callback_path.rs

```diff
@@ -0,0 +1,54 @@
+use crate::prelude::*;
+
+#[derive(
+    Clone,
+    PartialEq,
+    Eq,
+    Debug,
+    derive_more::FromStr,
+    derive_more::Display,
+    DeserializeFromStr,
+    SerializeDisplay,
+)]
+pub struct RCMCallbackPath(pub(crate) String);
+
+impl RCMCallbackPath {
+    pub fn new(value: impl Into<String>) -> Self {
+        Self(value.into())
+    }
+}
+
+impl Default for RCMCallbackPath {
+    fn default() -> Self {
+        Self("default_callback_path".to_owned())
+    }
+}
+
+impl HasSampleValues for RCMCallbackPath {
+    fn sample() -> Self {
+        Self("sample_callback_path".to_owned())
+    }
+
+    fn sample_other() -> Self {
+        Self("sample_callback_path_other".to_owned())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[allow(clippy::upper_case_acronyms)]
+    type SUT = RCMCallbackPath;
+
+    #[test]
+    fn equality() {
+        assert_eq!(SUT::sample(), SUT::sample());
+        assert_eq!(SUT::sample_other(), SUT::sample_other());
+    }
+
+    #[test]
+    fn inequality() {
+        assert_ne!(SUT::sample(), SUT::sample_other());
+    }
```

Comment:
add JSON tests

Reply:
JSON tests are performed in the dapp_definitions for a whole model, is there a need for them here too?

Reply:
takes 1 min to add them, so yes please do, it asserts that the Serde behaviour of this newtype is indeed "transparent", that wrapping the str in this newtype does not add one layer.

If the test fails you need to add `#[serde(transparent)]`, see `BIP39Passphrase`